File: stick-rumble-client/src/game/entities/WeaponGeometry.ts
export function getWeaponBarrelLength(weaponType: string): number {
  switch (weaponType.toLowerCase()) {
    case 'uzi':
      return 26;
    case 'ak47':
      return 40;
    case 'shotgun':
      return 37;
    case 'bat':
      return 40;
    case 'katana':
      return 55;
    case 'pistol':
    default:
      return 25;
  }
}

export function getWeaponBarrelTipPosition(
  weaponX: number,
  weaponY: number,
  aimAngle: number,
  weaponType: string
): { x: number; y: number } {
  const barrelLength = getWeaponBarrelLength(weaponType);

  return {
    x: weaponX + Math.cos(aimAngle) * barrelLength,
    y: weaponY + Math.sin(aimAngle) * barrelLength,
  };
}

File: stick-rumble-server/internal/game/weapon_geometry.go
package game

import "math"

const weaponGripOffset = 10.0

func getWeaponBarrelLength(weaponType string) float64 {
	switch weaponType {
	case "Uzi", "uzi":
		return 36
	case "AK47", "ak47":
		return 50
	case "Shotgun", "shotgun":
		return 47
	case "Bat", "bat":
		return 50.5
	case "Katana", "katana":
		return 65
	case "Pistol", "pistol":
		fallthrough
	default:
		return 25
	}
}

func getWeaponFireOrigin(playerPos Vector2, aimAngle float64, weaponType string) Vector2 {
	offset := weaponGripOffset + getWeaponBarrelLength(weaponType)
	return Vector2{
		X: playerPos.X + math.Cos(aimAngle)*offset,
		Y: playerPos.Y + math.Sin(aimAngle)*offset,
	}
}
